### PR TITLE
[SHIPA-1943] Makes CName schemes configurable

### PR DESCRIPTION
**Pull request by stinkyfingers**
# Description

Makes the http/https scheme configurable per Cname. Switches `app.ingress.cnames` from `[]string` to `[]Cname{Name: <string>, Secure: <bool>}`. 
This change is backward compatible with:
- Using the CLI. Cnames were added with the `ketch cname add` command. I added a `--secure` flag. It will return a user error if you specify `secure` and the app's framework does not have a clusterIssuer. 
- Using app.yamls. Deploying a ketch app by yaml only permits **one** cname (probably worth changing in a new issue). I added a `secure` field to the `cname` object. 
  ```
  cname:
    dnsName: secure.10.96.13.219.shipa.cloud
    secure: true
  ```
  It will return user error if you specify `secure` and a framework without a cluster issuer. 

This PR changes what users will supply in a kubernetes config file. Cnames are an object rather than string. E.g. 
```
apiVersion: theketch.io/v1beta1
kind: App
metadata:
  name: test
  namespace: foobar
spec:
  canary: {}
  deployments:
  - image: gcr.io/kubernetes-312803/sample-go-app:latest
    processes:
    - cmd:
      - web
      name: web
      units: 1
    routingSettings:
      weight: 100
    version: 1
    labels:
      - name: test-label
        value: works
  deploymentsCount: 1
  dockerRegistry: {}
  framework: test
  ingress:
    generateDefaultCname: false
    cnames:
      - name: secure.10.96.13.219.shipa.cloud
        secure: true
      - name: insecure.10.96.13.219.shipa.cloud
        secure: false
```

Fixes # [1943](https://shipaio.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=SHIPA&modal=detail&selectedIssue=SHIPA-1943)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [x] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [ ] This change requires no testing (i.e. documentation update)

## Documentation

- [x] All added public packages, funcs, and types have been documented with doc comments
- [x] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [x] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: cmd/ketch/cname_add.go [hunk 1]
```diff
@@ -31,6 +33,8 @@ func newCnameAddCmd(cfg config, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().StringVarP(&options.appName, deploy.FlagApp, deploy.FlagAppShort, "", "The name of the app.")
 	cmd.MarkFlagRequired("app")
+	cmd.Flags().BoolVar(&options.secure, flagSecure, false, "Whether the CName should be https")
```
Comment (DavisFrench) [inline cmd/ketch/cname_add.go:36]: This is just a nit pick, but I feel there is not much of a need to have the flag name as a const.

**Review by DavisFrench — APPROVED** [review]
Looks good to me! I left one comment, but it is just a nit pick.

File: internal/api/v1beta1/app_types.go [hunk 2]
```diff
@@ -52,7 +52,12 @@ type Label struct {
 }
 
 // CnameList is a list of an app's CNAMEs.
-type CnameList []string
+type CnameList []Cname
+
+type Cname struct {
```
Comment (aleksej-paschenko) [inline internal/api/v1beta1/app_types.go:57]: it would be great to have godoc

**Review by aleksej-paschenko — APPROVED** [review]
looks awesome!

**Pull request merged**